### PR TITLE
Add DNS fields to Container and Network unit groups

### LINKED ISSUE #19884: support `--dns` in Quadlet
### Feature request description

At the time of writing, Quadlet does not support the `--dns` flag in form of a field.

Noticed while reviewing a blog post that is resorting to `PodmanArgs=--dns=...`.

### Suggest potential solution

Add a `DNS` field to `.container` filees.

### Have you considered any alternatives?

`PodmanArgs` is a work around but not idiomatic.

### Additional context

Cc: @ygalblum @rhatdan

### PR DESCRIPTION
**Pull request by ryanhockstad**
#### Does this PR introduce a user-facing change?

```release-note
Quadlet unit files allow DNS field in Network group and DNS, DNSSearch, and DNSOption field in Container group.
```

closes #19884

### CODE REVIEW

**Comment by TomSweeneyRedHat** [comment]
Quick late night browse LGTM, but definitely want less tired eyeballs on it.

**Comment by ygalblum** [comment]
Code LGTM, but the linter is complaining on trailing-whitespace in docs/source/markdown/podman-systemd.unit.5.md

File: pkg/systemd/quadlet/quadlet.go [hunk 1]
```diff
@@ -483,6 +490,21 @@ func ConvertContainer(container *parser.UnitFile, names map[string]string, isUse
...
 	}
 
+	dns := container.LookupAll(ContainerGroup, KeyDNS)
+	for _, ipAddr := range dns {
+		podman.addf("--dns=%s", ipAddr)
+	}
+
+	dnsOptions := container.LookupAll(ContainerGroup, KeyDNSOption)
+	for _, dnsOption := range dnsOptions {
+		podman.addf("--dns-search=%s", dnsOption)
```
Comment (Luap99) [inline pkg/systemd/quadlet/quadlet.go:500]: typo this must be --dns-option

Comment (ryanhockstad) [reply pkg/systemd/quadlet/quadlet.go]: Nice catch. Fixed this along with the trailing whitespace in the docs.

**Comment by Luap99** [comment]
Your new tests are failing

**Comment by edsantiago** [comment]
@ryanhockstad please rebase on 9554022ce to get CI working.

**Comment by mheon** [comment]
Code LGTM

**Comment by rhatdan** [comment]
/lgtm
/hold
Thanks @ryanhockstad

**Comment by ryanhockstad** [comment]
I think I fixed the tests I added that were failing. Should I squash the commit that merged the upstream into my branch? I sync'd my repo through the UI instead of rebasing.

**Pull request merged**